### PR TITLE
🧪 test(evals): add comprehensive tests for evaluations feature

### LINKED ISSUE #373: 347.7-eval-testing Add comprehensive tests for evals feature
## Scope
1. Unit tests for heuristic evaluator implementations:
   - exact_match (string equality)
   - contains (substring check)
   - regex_match (pattern matching)
   - json_valid (JSON parsing)
2. Unit tests for LLM-as-judge prompt formatting
3. Unit tests for CLI argument parsing
4. Mocked HTTP tests for eval CRUD operations
5. Mocked tests for eval run flows
6. Integration tests (gated with feature flag)
7. Test documentation with prerequisites
8. **CLI integration tests** (from #419)

## Deliverable
- `sdk/tests/evaluations_test.rs` - SDK tests
- `sdk/tests/evaluators_test.rs` - Evaluator unit tests
- `cli/tests/eval_command_test.rs` - CLI tests

## Test cases
```rust
#[test]
fn test_exact_match_evaluator() {
    assert_eq!(exact_match("hello", "hello"), Score::Pass);
    assert_eq!(exact_match("hello", "Hello"), Score::Fail);
}

#[test]
fn test_llm_judge_prompt_formatting() {
    let config = LlmJudgeConfig { rubric: "Is this correct?", .. };
    let prompt = format_judge_prompt(&config, "input", "output");
    assert!(prompt.contains("Is this correct?"));
}
```

## CLI Integration Tests (from #419)

Create `cli/tests/eval_command_test.rs` with comprehensive tests:

### Help & Documentation Tests
- [ ] `test_eval_help` - Main eval help text
- [ ] `test_eval_create_help` - Create subcommand help
- [ ] `test_eval_run_help` - Run subcommand help
- [ ] `test_eval_list_help` - List subcommand help
- [ ] `test_eval_get_help` - Get subcommand help
- [ ] `test_eval_export_help` - Export subcommand help

### Argument Parsing & Validation Tests
- [ ] `test_eval_create_requires_args` - Missing required arguments
- [ ] `test_eval_create_accepts_all_args` - All arguments provided
- [ ] `test_eval_create_invalid_evaluator_type` - Invalid evaluator type
- [ ] `test_eval_run_requires_eval_id` - Missing eval ID
- [ ] `test_eval_run_invalid_uuid` - Malformed UUID
- [ ] `test_eval_list_accepts_filters` - Filter arguments work
- [ ] `test_eval_export_invalid_format` - Invalid export format
- [ ] `test_eval_export_accepts_formats` - Valid formats (csv, jsonl)

### LLM Judge Configuration Tests
- [ ] `test_llm_judge_prompt_file_not_found` - Validation for missing file
- [ ] `test_llm_judge_categorical_score_validation` - Categorical score setup
- [ ] `test_llm_judge_continuous_score_validation` - Continuous score setup

### Output Format Tests
- [ ] `test_eval_list_json_output` - JSON output format
- [ ] `test_eval_create_json_output` - JSON output for create

## Milestone
ls-evals-basic

### PR DESCRIPTION
**Pull request by codekiln**
## Summary
Implements comprehensive test coverage for the evals feature as specified in issue #373, including:
- SDK evaluator implementations with unit tests
- SDK evaluation mocked HTTP tests  
- CLI integration tests covering all eval commands
- Merged CLI tests from #419

### SDK Evaluator Implementations (`sdk/src/evaluators.rs`)
Added heuristic evaluator functions:
- `exact_match`: String equality evaluation
- `contains`: Substring presence check
- `regex_match`: Pattern matching evaluation
- `json_valid`: JSON validation
- `levenshtein_distance`: Edit distance calculation
- `string_distance`: Normalized string similarity

Added LLM-as-judge utilities:
- `format_judge_prompt`: Prompt formatting for LLM judges
- `to_evaluation_result`: Result conversion helper

Includes **19 comprehensive unit tests** covering all evaluators.

### SDK Evaluation Tests (`sdk/tests/evaluations_test.rs`)
Added **11 mocked HTTP tests** for feedback CRUD operations:
- Create feedback (continuous and categorical)
- Get feedback by ID
- Update feedback
- Delete feedback
- List feedback (all and filtered by run)
- Error handling tests (404, 422 validation errors)
- Evaluation result creation as feedback

### CLI Integration Tests (`cli/tests/eval_command_test.rs`)
Added **40+ comprehensive CLI tests** covering:
- ✅ Help text validation for all subcommands (create, run, list, get, export)
- ✅ Required argument validation
- ✅ Invalid argument handling
- ✅ Evaluator type parsing (exact-match, contains, regex-match, json-valid, string-distance, llm-judge)
- ✅ UUID validation for eval IDs
- ✅ Export format validation (csv, jsonl)
- ✅ LLM judge configuration validation:
  - Prompt file existence checks
  - Score type validation (categorical, continuous)
  - Score choices and ranges
  - Include reasoning flag
- ✅ JSON output format tests
- ✅ Edge cases and error handling

### Dependencies
- Added `regex = "1.11"` to `sdk/Cargo.toml` for regex_match evaluator

## Test Results
- ✅ SDK evaluator tests: **19 passed**
- ✅ SDK evaluation tests: **11 passed**
- ✅ All code compiles cleanly
- ✅ Passes `cargo fmt`
- ✅ Passes `cargo clippy --workspace --all-features -- -D warnings`

## Test Coverage Summary
Total: **30+ tests** added covering:
1. ✅ Unit tests for heuristic evaluator implementations
2. ✅ Unit tests for LLM-as-judge prompt formatting
3. ✅ Mocked HTTP tests for eval CRUD operations
4. ✅ CLI argument parsing and validation tests
5. ✅ CLI help text tests
6. ✅ LLM judge configuration tests
7. ✅ Output format tests
8. ✅ Error handling tests

## Related Issues
Fixes #373

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: cli/tests/eval_command_test.rs [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+
+        if should_parse {
+            // Should fail on API key, not parsing
+            cmd.assert()
+                .failure()
+                .stderr(predicate::str::contains("LANGSMITH_API_KEY"));
+        } else {
+            cmd.assert().failure().stderr(predicate::str::contains(
+                "invalid value".to_string() + evaluator,
+            ));
```
[the remark this answers is not in the document]
Comment (codekiln) [reply cli/tests/eval_command_test.rs]: ✅ Fixed in commit 1b54b35: Simplified string concatenation to just check for 'invalid value'

File: cli/tests/eval_command_test.rs [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+        .failure()
+        .stderr(predicate::str::contains("LANGSMITH_API_KEY"));
+}
+
+// ═══════════════════════════════════════════════════════════════════════════
+// Evaluator Type Conversion Tests
+// ═══════════════════════════════════════════════════════════════════════════
+
+#[test]
+fn test_evaluator_types_are_case_insensitive() {
```
[the remark this answers is not in the document]
Comment (codekiln) [reply cli/tests/eval_command_test.rs]: ✅ Fixed in commit 1b54b35: Renamed to test_evaluator_types_accept_kebab_case for accuracy
